lib/param_parsing/parameters.go

```diff
@@ -56,17 +56,20 @@ func GetParameters() Params {
 	logger = lib.InitLogger(params.LogLevel)
 	var err error
 	// Read configuration files
-	if tomlFileExists(params) {
-		params, err = getParamsTOML(params)
-	} else if tfSwitchFileExists(params) {
+	params, err = getParamsTOML(params)
+	if params.Version == "" {
```

Comment:
This feels fine*ish* if we _know_ the only param value we are obtaining is `Version`.
In the (unlikely?) event that we wish to obtain additional params from these, perhaps a better way is to obtain in order of least-to-highest priority, without the conditions, allowing each higher priority method to override the values from the previous. But, as long as we only care about `Version`, this is probably a more efficient method.

Reply:
I'm also wondering if it's nicer to move the "obtain version" to a separate function, which calls (e.g. `GetVersionFromTerragrunt`), changing these GetVersion* methods to only return a version strring and then return-fast, e.g.:
```
func GetParams() {
...
  params = getParamsToml(params)
  if params.Version == "" {
     params.Version = getParamsFromSources(params)
   }
...
}

func getParamsFromSources(params) string {
    version := ""
    if version = GetVersionFromTfSwitch(params); version != "" {
        return version
    }
    if version = GetParamsFromTerraformVersion(params); version != "" {
        return version
    }
    ...
    return ""
}
```

Reply:
Continued this discussion as it may play a role in https://github.com/warrensbox/terraform-switcher/discussions/420#discussioncomment-9364783